### PR TITLE
Add new versions of ectrans, eckit, fckit, atlas; remove py-h5py from ufs-pyenv

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Add new versions of `ectrans`, `eckit`, `fckit`, `ecmwf-atlas`. Use `sha256` where possible, also update `fiat` package to be able to use `sha256` in the future.

Remove `py-h5py` from `ufs-pyenv`. To my knowledge, this isn't used. The only piece of software that uses `py-h5py` is JEDI's `ioda-converters` in the JEDI package, and therefore `py-h5py` is in `jedi-base-env`. @AlexanderRichert-NOAA @Hang-Lei-NOAA can you confirm?

## Testing

Tested with https://github.com/NOAA-EMC/spack-stack/pull/408.

### CODE REVIEW

File: var/spack/repos/builtin/packages/eckit/package.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ class Eckit(CMakePackage):
 
     maintainers = ["skosukhin", "climbfuji"]
 
+    version("1.20.2", sha256="779be536e22f985b00927330d8430794c43f4f48")
+    version("1.20.0", sha256="5ecebc9548eba8df1c3e6f162e8f480257528eab")
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/builtin/packages/eckit/package.py]: These don't seem to be valid git commit hashes or sha256 sums

File: var/spack/repos/jcsda-emc-bundles/packages/ufs-pyenv/package.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class UfsPyenv(BundlePackage):
 
     depends_on("py-cython")
     depends_on("py-cftime")
-    depends_on("py-h5py")
+    #depends_on("py-h5py")
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/jcsda-emc-bundles/packages/ufs-pyenv/package.py:23]: I don't see h5py getting used anywhere in UFS, at least I don't see any explicit import's.

File: var/spack/repos/jcsda-emc/packages/ecmwf-atlas/package.py [hunk 3]
```diff
@@ -13,11 +13,13 @@ class EcmwfAtlas(CMakePackage):
...
     url = "https://github.com/ecmwf/atlas/archive/0.22.1.tar.gz"
 
-    maintainers = ["climbfuji", "rhoneyager"]
+    maintainers = ["climbfuji", "srherbener"]
 
     version("master", branch="master")
     version("develop", branch="develop")
-    version("0.30.0", commit="d81887206fc49c83fd5eb2e0e17457bdcd04731a", preferred=True)
+    version("0.31.1", sha256="0e14c9512235c1f66e10c25e190cfeb32275c2be")
+    version("0.31.0", sha256="954f0f06991c48dfa25c7f162427e14970085f3d")
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/jcsda-emc/packages/ecmwf-atlas/package.py]: These don't seem to be valid git commit hashes or sha256 sums

File: var/spack/repos/jcsda-emc/packages/ectrans/package.py [hunk 4]
```diff
@@ -15,13 +15,15 @@ class Ectrans(CMakePackage):
 
     homepage = "https://github.com/ecmwf-ifs/ectrans"
     git = "https://github.com/ecmwf-ifs/ectrans.git"
+    url = "https://github.com/ecmwf-ifs/ectrans/archive/1.1.0.tar.gz"
 
     maintainers = ['climbfuji']
 
     version('develop', branch='develop', no_cache=True)
     version('main', branch='main', no_cache=True)
+    version("1.1.0", sha256="bc287604cd36ca32618f21767f4a8b85249e3506")
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/jcsda-emc/packages/ectrans/package.py]: This doesn't seem to be a valid git commit hash or sha256 sum

File: var/spack/repos/builtin/packages/eckit/package.py (hunk 1 above)
Comment (climbfuji) [reply var/spack/repos/builtin/packages/eckit/package.py]: You are right, they are much shorter but for whatever reason they worked just fine. I'll replace them with a made-up hash of the right length and see what happens. Same for the other occurrences below.

**Review by srherbener — APPROVED** [review]
I think the changes look good once the questions about the sha256 hashes are resolved.

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA The sha256 checksums should now be fixed.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
CI tests passed (except one of the skylab builds running out of time at 6hrs) for https://github.com/NOAA-EMC/spack-stack/pull/408/commits/699160254bc9c12e331a957307ba39c6e24a73dd. Going to merge this.

**Pull request merged**